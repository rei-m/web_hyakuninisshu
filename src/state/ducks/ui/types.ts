import { Action } from 'redux';
import { Color, Kimariji } from '@src/types';
import * as constants from './constants';

export type OPEN_KARUTAS_FILTER_TYPE = typeof constants.OPEN_KARUTAS_FILTER_NAME;

export type CLOSE_KARUTAS_FILTER_TYPE = typeof constants.CLOSE_KARUTAS_FILTER_NAME;

export type TOGGLE_KARUTAS_COLOR_TYPE = typeof constants.TOGGLE_KARUTAS_COLOR_NAME;

export type TOGGLE_KARUTAS_KIMARIJI_TYPE = typeof constants.TOGGLE_KARUTAS_KIMARIJI_NAME;

export interface OpenKarutasFilterAction extends Action {
  readonly type: OPEN_KARUTAS_FILTER_TYPE;
}

export interface CloseKarutasFilterAction extends Action {
  readonly type: CLOSE_KARUTAS_FILTER_TYPE;
}

export interface OKarutasFilterAction extends Action {
  readonly type: OPEN_KARUTAS_FILTER_TYPE;
}

export interface ToggleKarutasColorAction extends Action {
  readonly type: TOGGLE_KARUTAS_COLOR_TYPE;
  readonly payload: {
    readonly color: Color;
    readonly checked: boolean;
  };
}

export interface ToggleKarutasKimarijiAction extends Action {
  readonly type: TOGGLE_KARUTAS_KIMARIJI_TYPE;
  readonly payload: {
    readonly kimariji: Kimariji;
    readonly checked: boolean;
  };
}

export type Actions =
  | OpenKarutasFilterAction
  | CloseKarutasFilterAction
  | ToggleKarutasColorAction
  | ToggleKarutasKimarijiAction;

export interface KarutasFilter {
  readonly colors: Array<{ color: Color; checked: boolean }>;
  readonly kimarijis: Array<{ kimariji: Kimariji; checked: boolean }>;
}

export interface UiState {
  readonly karutasFilter: KarutasFilter & { readonly open: boolean };
}
