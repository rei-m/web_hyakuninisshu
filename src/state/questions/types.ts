import { Action } from 'redux';
import { Kimariji, Color, QuestionId, KarutaNo, Karuta } from '@src/domain/models';
import * as constants from './constants';
import { Meta, Payload } from '..';

type Condition<T> = {
  values: Array<T>;
};

export type RangeFromCondition = 1 | 11 | 21 | 31 | 41 | 51 | 61 | 71 | 81 | 91;
export const RangeFromCondition: Condition<RangeFromCondition> = {
  values: [1, 11, 21, 31, 41, 51, 61, 71, 81, 91],
};

export type RangeToCondition = 10 | 20 | 30 | 40 | 50 | 60 | 70 | 80 | 90 | 100;
export const RangeToCondition: Condition<RangeToCondition> = {
  values: [10, 20, 30, 40, 50, 60, 70, 80, 90, 100],
};

export type KimarijiCondition = null | Kimariji;
export const KimarijiCondition: Condition<KimarijiCondition> = {
  values: [null, 1, 2, 3, 4, 5, 6],
};

export type ColorCondition = null | Color;
export const ColorCondition: Condition<ColorCondition> = {
  values: [null, 'blue', 'pink', 'yellow', 'green', 'orange'],
};

export type KarutaStyleCondition = 'kanji' | 'kana';
export const KarutaStyleCondition: Condition<KarutaStyleCondition> = {
  values: ['kanji', 'kana'],
};

export type QuestionAnimCondition = 'none' | 'slow' | 'normal' | 'fast';
export const QuestionAnimCondition: Condition<QuestionAnimCondition> & {
  toDulation: (value: QuestionAnimCondition) => number;
} = {
  values: ['none', 'slow', 'normal', 'fast'],
  toDulation: (value: QuestionAnimCondition) => {
    switch (value) {
      case 'normal':
        return 0.6;
      case 'fast':
        return 0.3;
      case 'slow':
        return 1.0;
      default:
        return 0;
    }
  },
};

export type YomiFuda = Readonly<{
  karutaNo: KarutaNo;
  shoku: string;
  niku: string;
  sanku: string;
}>;

export type ToriFuda = Readonly<{
  karutaNo: KarutaNo;
  shiku: string;
  kekku: string;
}>;

export type StartTrainingAction = Action<typeof constants.START_TRAINING_NAME> &
  Payload<{
    currentQuestionId: QuestionId;
    totalCount: number;
  }> &
  Meta<{
    rangeFrom: RangeFromCondition;
    rangeTo: RangeToCondition;
    kimariji: KimarijiCondition;
    color: ColorCondition;
    kamiNoKuStyle: KarutaStyleCondition;
    shimoNoKuStyle: KarutaStyleCondition;
    questionAnim: QuestionAnimCondition;
  }>;

export type RestartTrainingAction = Action<typeof constants.RESTART_TRAINING_NAME> &
  Payload<{
    currentQuestionId: QuestionId;
    totalCount: number;
  }>;

export type StartExamAction = Action<typeof constants.START_EXAM_NAME> &
  Payload<{
    currentQuestionId: QuestionId;
    totalCount: number;
  }>;

export type StartQuestionAction = Action<typeof constants.START_QUESTION_NAME> &
  Payload<{
    questionId: QuestionId;
    content: {
      yomiFuda: YomiFuda;
      toriFudaList: [ToriFuda, ToriFuda, ToriFuda, ToriFuda];
    };
  }>;

export type ResetQuestionAction = Action<typeof constants.RESET_QUESTION_NAME>;

export type FinishQuestionAction = Action<typeof constants.FINISH_QUESTION_NAME> &
  Payload<{
    correctCount: number;
    averageAnswerSecond: number;
    answerList: Array<{
      questionId: QuestionId;
      isCorrect: boolean;
      correctKaruta: Karuta;
    }>;
  }>;

export type AnswerQuestionAction = Action<typeof constants.ANSWER_QUESTION_NAME> &
  Payload<{
    isCorrect: boolean;
    selectedKarutaNo: KarutaNo;
    correctKaruta: Karuta;
  }>;

export type ConfirmCorrectAction = Action<typeof constants.CONFIRM_CORRECT_NAME>;

export type OpenNextQuestionAction = Action<typeof constants.OPEN_NEXT_QUESTION_NAME> &
  Payload<{
    currentQuestionId: QuestionId;
  }>;

export type Actions =
  | StartTrainingAction
  | RestartTrainingAction
  | StartExamAction
  | StartQuestionAction
  | ResetQuestionAction
  | FinishQuestionAction
  | AnswerQuestionAction
  | ConfirmCorrectAction
  | OpenNextQuestionAction;

export type ActionCreator = {
  /**
   * 指定した条件から問題を作成/保存してStartTrainingActionを返す
   *
   * @param rangeFrom 出題範囲From
   * @param rangeTo 出題範囲To
   * @param kimariji 決まり字条件
   * @param color 色条件
   * @param kamiNoKuStyle 上の句表示形式
   * @param shimoNoKuStyle 下の句表示形式
   * @param questionAnim 問題の表示速度
   *
   * @returns {StartTrainingAction}
   * @throws {Error} 指定した条件に当てはまる歌が無い場合
   */
  startTraining: (
    rangeFrom: RangeFromCondition,
    rangeTo: RangeToCondition,
    kimariji: KimarijiCondition,
    color: ColorCondition,
    kamiNoKuStyle: KarutaStyleCondition,
    shimoNoKuStyle: KarutaStyleCondition,
    questionAnim: QuestionAnimCondition
  ) => StartTrainingAction;

  /**
   * 回答済の問題の中から間違えた問題をもとに問題を作成/保存してRestartTrainingActionを返す
   *
   * @returns {RestartTrainingAction}
   * @throws {Error} 指定した条件に当てはまる歌が無い場合
   */
  restartTraining: () => RestartTrainingAction;

  /**
   * 全ての歌を対象に問題を作成/保存してStartExamActionを返す
   *
   * @returns {StartExamAction}
   * @throws {Error} 指定した条件に当てはまる歌が無い場合
   */
  startExam: () => StartExamAction;

  /**
   * 問題の回答を開始してStartQuestionActionを返す
   *
   * @returns {StartQuestionAction}
   * @throws {Error} 指定した問題がない場合
   */
  startQuestion: (
    currentQuestionId: QuestionId,
    kamiNoKuStyle: KarutaStyleCondition,
    shimoNoKuStyle: KarutaStyleCondition,
    startDate: Date
  ) => StartQuestionAction;

  /**
   * 問題の回答を判定してAnswerQuestionActionを返す
   *
   * @returns {AnswerQuestionAction}
   * @throws {Error} 指定した問題がない場合
   * @throws {Error} 問題の回答が開始されていない場合
   */
  answerQuestion: (currentQuestionId: QuestionId, toriFuda: ToriFuda, answerDate: Date) => AnswerQuestionAction;

  /**
   * ConfirmCorrectActionを返す
   *
   * @returns {ConfirmCorrectAction}
   */
  confirmCorrect: () => ConfirmCorrectAction;

  /**
   * 次の問題を取得してOpenNextQuestionActionを返す
   *
   * @returns {OpenNextQuestionAction}
   * @throws {Error} 次の問題がない場合
   */
  openNextQuestionAction: (currentQuestionId: QuestionId) => OpenNextQuestionAction;

  /**
   * 問題を初期状態に戻してResetQuestionActionを返す
   *
   * @returns {ResetQuestionAction}
   */
  resetQuestion: () => ResetQuestionAction;

  /**
   * 問題を終了して結果を見る
   *
   * @returns {FinishQuestionAction}
   */
  finishQuestion: () => FinishQuestionAction;
};

export type State = Readonly<{
  state: 'waiting' | 'ready' | 'playing' | 'confirm' | 'finished';
  totalCount: number;
  currentPosition: number;
  currentQuestion?: {
    questionId: QuestionId;
    content?: {
      yomiFuda: YomiFuda;
      toriFudaList: [ToriFuda, ToriFuda, ToriFuda, ToriFuda];
    };
    answer?: {
      isCorrect: boolean;
      selectedKarutaNo: KarutaNo;
      correctKaruta: Karuta;
    };
  };
  result?: {
    correctCount: number;
    averageAnswerSecond: number;
    answerList: Array<{
      questionId: QuestionId;
      isCorrect: boolean;
      correctKaruta: Karuta;
    }>;
  };
  trainingCondition: {
    rangeFrom: RangeFromCondition;
    rangeTo: RangeToCondition;
    kimariji: KimarijiCondition;
    color: ColorCondition;
    kamiNoKuStyle: KarutaStyleCondition;
    shimoNoKuStyle: KarutaStyleCondition;
    questionAnim: QuestionAnimCondition;
  };
}>;
