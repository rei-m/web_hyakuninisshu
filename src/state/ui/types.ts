import { Action } from 'redux';
import * as constants from './constants';
import { Color, Kimariji } from '@src/domain/models';
import { Payload } from '..';

export type KarutasFilter = Readonly<{
  colors: Array<{ color: Color; checked: boolean }>;
  kimarijis: Array<{ kimariji: Kimariji; checked: boolean }>;
}>;

export type OpenKarutasFilterAction = Action<typeof constants.OPEN_KARUTAS_FILTER_NAME>;

export type CloseKarutasFilterAction = Action<typeof constants.CLOSE_KARUTAS_FILTER_NAME>;

export type ToggleKarutasColorAction = Action<typeof constants.TOGGLE_KARUTAS_COLOR_NAME> &
  Payload<{
    color: Color;
    checked: boolean;
  }>;

export type ToggleKarutasKimarijiAction = Action<typeof constants.TOGGLE_KARUTAS_KIMARIJI_NAME> &
  Payload<{
    kimariji: Kimariji;
    checked: boolean;
  }>;

export type Actions =
  | OpenKarutasFilterAction
  | CloseKarutasFilterAction
  | ToggleKarutasColorAction
  | ToggleKarutasKimarijiAction;

export type ActionCreator = {
  /**
   * OpenKarutasFilterActionを返す
   *
   * @returns {OpenKarutasFilterAction}
   */
  openKarutasFilter: () => OpenKarutasFilterAction;

  /**
   * CloseKarutasFilterActionを返す
   *
   * @returns {CloseKarutasFilterAction}
   */
  closeKarutasFilter: () => CloseKarutasFilterAction;

  /**
   * ToggleKarutasColorActionを返す
   *
   * @param color 歌の色
   * @param checked 表示する/しない
   *
   * @returns {ToggleKarutasColorAction}
   */
  toggleKarutasColor: (color: Color, checked: boolean) => ToggleKarutasColorAction;

  /**
   * ToggleKarutasKimarijiActionを返す
   *
   * @param kimariji 歌の決まり字
   * @param checked 表示する/しない
   *
   * @returns {ToggleKarutasKimarijiAction}
   */
  toggleKarutasKimariji: (kimariji: Kimariji, checked: boolean) => ToggleKarutasKimarijiAction;
};

export type State = Readonly<{
  karutasFilter: KarutasFilter & { open: boolean };
}>;
