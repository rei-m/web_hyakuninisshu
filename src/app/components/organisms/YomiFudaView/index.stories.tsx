import { Meta, StoryObj } from '@storybook/react';
import YomiFudaView from './index';
import { KARUTA_LIST } from '@/assets/karuta';

export default {
  title: 'organisms/YomiFudaView',
  component: YomiFudaView,
} as Meta<typeof YomiFudaView>;

type Story = StoryObj<typeof YomiFudaView>;

const karuta = KARUTA_LIST[0];

export const Default: Story = {
  args: {
    yomiFuda: {
      karutaNo: karuta.no,
      shoku: karuta.shoku.kanji,
      niku: karuta.niku.kanji,
      sanku: karuta.sanku.kanji,
    },
    answered: false,
    duration: 1000,
  },
};
