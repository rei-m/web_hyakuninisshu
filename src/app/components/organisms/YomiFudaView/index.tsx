import type { YomiFuda } from '@/domains/models';
import type { SxAppProps } from '@/theme';

import { useState } from 'react';

import Box from '@mui/material/Box';
import Phrase, { SPACE } from './Phrase';
import { FONT_SIZE } from '@/theme';

type Size = 's' | 'm' | 'l';

const RATIO_L = 1.15;
const RATIO_M = 1.0;
const RATIO_S = 0.875;
const RATIO_MAP = {
  s: RATIO_S,
  m: RATIO_M,
  l: RATIO_L,
} as const;

export type YomiFudaViewProps = {
  yomiFuda: YomiFuda;
  answered: boolean;
  duration: number;
  size?: Size;
  sx?: SxAppProps;
};

export type YomiFudaViewPresenterProps = {
  firstLine: string;
  secondLine: string;
  thirdLine: string;
  duration: number;
  size?: Size;
  sx?: SxAppProps;
  onAnimationEnd: () => void;
};

export type YomiFudaViewContainerProps = YomiFudaViewProps & { presenter: React.FC<YomiFudaViewPresenterProps> };

const adjustDisplayText = (text: string, startIndex: number, currentPosition: number) => {
  if (currentPosition < startIndex) {
    return Array.from(Array(text.length).keys())
      .map(() => SPACE)
      .join('');
  }
  const line = currentPosition > startIndex ? text.substring(0, currentPosition - startIndex) : '';
  const mod = text.length - (currentPosition - startIndex);
  const linePad =
    mod > 0
      ? Array.from(Array(mod).keys())
          .map(() => SPACE)
          .join('')
      : '';
  return line + linePad;
};

export const YomiFudaPresenter = ({
  firstLine,
  secondLine,
  thirdLine,
  duration,
  size = 'm',
  sx,
  onAnimationEnd,
}: YomiFudaViewPresenterProps) => (
  <Box
    sx={[
      {
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: '#fffff0',
        borderStyle: 'solid',
        borderColor: 'primary.dark',
        borderWidth: `${5 * RATIO_MAP[size]}px`,
        borderRadius: 2,
        width: `${120 * RATIO_MAP[size]}px`,
        height: `${205 * RATIO_MAP[size]}px`,
      },
      ...(Array.isArray(sx) ? sx : [sx]),
    ]}
  >
    <Box
      sx={{
        display: 'flex',
        flexDirection: 'row-reverse',
        justifyContent: 'center',
      }}
    >
      <Phrase text={firstLine} duration={duration} fontSize={FONT_SIZE[size]} onAnimationEnd={onAnimationEnd} />
      <Phrase
        text={secondLine}
        duration={duration}
        fontSize={FONT_SIZE[size]}
        onAnimationEnd={onAnimationEnd}
        sx={{
          paddingTop: `${8 * 3 * RATIO_MAP[size]}px`,
          marginLeft: `${8 * RATIO_MAP[size]}px`,
          marginRight: `${8 * RATIO_MAP[size]}px`,
        }}
      />
      <Phrase
        text={thirdLine}
        duration={duration}
        fontSize={FONT_SIZE[size]}
        onAnimationEnd={onAnimationEnd}
        sx={{
          paddingTop: `${8 * 6 * RATIO_MAP[size]}px`,
        }}
      />
    </Box>
  </Box>
);

export const YomiFudaContainer = ({
  yomiFuda,
  answered,
  duration,
  size,
  sx,
  presenter,
}: YomiFudaViewContainerProps) => {
  const { shoku, niku, sanku } = yomiFuda;
  const durationOrAnswered = answered ? 0 : duration;

  const [position, setPosition] = useState(duration === 0 ? shoku.length + niku.length + sanku.length : 1);

  const firstLine = adjustDisplayText(shoku, 0, position);
  const secondLine = adjustDisplayText(niku, shoku.length, position);
  const thirdLine = adjustDisplayText(sanku, shoku.length + niku.length, position);

  const onAnimationEnd = () => {
    if (answered) {
      return;
    }
    setPosition(position + 1);
  };

  return presenter({
    firstLine,
    secondLine,
    thirdLine,
    duration: durationOrAnswered,
    size,
    sx,
    onAnimationEnd,
  });
};

const YomiFudaView = (props: YomiFudaViewProps) => <YomiFudaContainer presenter={YomiFudaPresenter} {...props} />;

export default YomiFudaView;
