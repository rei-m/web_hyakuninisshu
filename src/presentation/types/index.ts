import { RouteComponentProps } from '@reach/router';

export type SiteMetaData = {
  title: string;
  description: string;
  author: string;
};

export type GeneratedPageComponentProps<T = {}> = { pageContext: T } & RouteComponentProps<T>;

export type AppErrorType = 'Network' | 'NotFound' | 'Unknown';

export class AppError implements Error {
  private _message: string;
  private _type: AppErrorType;

  constructor(message: string, type: AppErrorType) {
    this._message = message;
    this._type = type;
  }

  public get name() {
    return 'AppError';
  }

  public get message() {
    return this._message;
  }

  public get type() {
    return this._type;
  }

  public toString() {
    return `${this.name}: ${this.message}`;
  }
}
