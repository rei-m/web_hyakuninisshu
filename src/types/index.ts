export interface Karuta {
  id: number;
  imageNo: string;
}

export interface Question {
  correctKaruta: Karuta;
  choices: Karuta[];
}

export interface OptionItem<T> {
  readonly name: string;
  readonly value: T;
}
