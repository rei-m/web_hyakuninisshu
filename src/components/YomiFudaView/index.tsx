import * as React from 'react';
import styled from '@src/styles/styled-components';
import { YomiFuda } from '@src/types';
import YomiFudaWord, { SPACE } from '@src/components/YomiFudaWord';

export interface Props {
  yomiFuda: YomiFuda;
  answered: boolean;
  style?: React.CSSProperties;
}

export interface State {
  currentPosition: number;
}

const Container = styled.div`
  width: 140px;
  height: 240px;
  border: 6px solid ${({ theme }) => theme.colorPrimaryDark};
  border-radius: 10px;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: ${({ theme }) => theme.colorThin};
  box-shadow: 0 -1px 5px rgba(0, 0, 0, 0.26);
  font-family: 'Sawarabi Mincho';
`;

const Inner = styled.div`
  display: flex;
  flex-direction: row-reverse;
  justify-content: center;
`;

const Phrase = styled.div`
  width: 1.9rem;
  font-size: 1.9rem;
  line-height: 2rem;
`;

const SecondPhrase = styled(Phrase)`
  padding-top: 24px;
  margin-left: 8px;
  margin-right: 8px;
`;

const ThirdPhrase = styled(Phrase)`
  padding-top: 48px;
`;

// 各フレーズの文字数を最大に合わせる
// 縦書きのエレメントを絶対位置で指定する

class YomiFudaView extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = { currentPosition: 1 };
    this.onAnimationEnd = this.onAnimationEnd.bind(this);
  }

  public render() {
    const { yomiFuda, style, answered } = this.props;
    const { firstText, secondText, thirdText } = yomiFuda;
    const firstLine = this.adjustDisplayText(firstText, 0);
    const secondLine = this.adjustDisplayText(secondText, firstText.length);
    const thirdLine = this.adjustDisplayText(thirdText, firstText.length + secondText.length);
    return (
      <Container style={style}>
        <Inner>
          {answered ? (
            <>
              <Phrase>{yomiFuda.firstText}</Phrase>
              <SecondPhrase>{yomiFuda.secondText}</SecondPhrase>
              <ThirdPhrase>{yomiFuda.thirdText}</ThirdPhrase>
            </>
          ) : (
            <>
              <Phrase>
                {Array.from(firstLine).map((s, i) => (
                  <YomiFudaWord word={s} onAnimationEnd={this.onAnimationEnd} key={i} />
                ))}
              </Phrase>
              <SecondPhrase>
                {Array.from(secondLine).map((s, i) => (
                  <YomiFudaWord word={s} onAnimationEnd={this.onAnimationEnd} key={i} />
                ))}
              </SecondPhrase>
              <ThirdPhrase>
                {Array.from(thirdLine).map((s, i) => (
                  <YomiFudaWord word={s} onAnimationEnd={this.onAnimationEnd} key={i} />
                ))}
              </ThirdPhrase>
            </>
          )}
        </Inner>
      </Container>
    );
  }

  private adjustDisplayText(text: string, startIndex: number) {
    const { currentPosition } = this.state;
    if (currentPosition < startIndex) {
      return Array.from(Array(text.length).keys())
        .map(_ => SPACE)
        .join('');
    }
    const line = currentPosition > startIndex ? text.substr(0, currentPosition - startIndex) : '';
    const mod = text.length - (currentPosition - startIndex);
    const linePad =
      mod > 0
        ? Array.from(Array(mod).keys())
            .map(_ => SPACE)
            .join('')
        : '';
    return line + linePad;
  }

  private onAnimationEnd() {
    this.setState({ currentPosition: this.state.currentPosition + 1 });
  }
}

export default YomiFudaView;
