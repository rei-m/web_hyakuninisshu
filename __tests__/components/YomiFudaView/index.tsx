import * as ReactTestRenderer from 'react-test-renderer';
import * as React from 'react';
import { create } from '@test/factories';
import YomiFudaView, { Props } from '@src/components/YomiFudaView';
import { YomiFuda } from '@src/types';

describe('<YomiFudaView />', () => {
  it('should render component with dulation', () => {
    const props: Props = {
      yomiFuda: create<YomiFuda>('yomiFuda'),
      answered: true,
      dulation: 0.6,
    };
    const renderer = ReactTestRenderer.create(<YomiFudaView {...props} />);
    expect(renderer.toJSON()).toMatchSnapshot();
  });

  it('should render component without dulation', () => {
    const props: Props = {
      yomiFuda: create<YomiFuda>('yomiFuda'),
      answered: true,
      dulation: 0,
    };
    const renderer = ReactTestRenderer.create(<YomiFudaView {...props} />);
    expect(renderer.toJSON()).toMatchSnapshot();
  });
});
