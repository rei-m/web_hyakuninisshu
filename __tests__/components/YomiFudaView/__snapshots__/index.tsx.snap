// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<YomiFudaView /> should render component 1`] = `
<div
  className="sc-bwzfXH bvmjNt"
>
  <div
    className="sc-htpNat eaelvy"
  >
    <div
      className="sc-bxivhb cIIvqJ"
    >
      初句
    </div>
    <div
      className="sc-bxivhb sc-ifAKCX ejiZuo"
    >
      二句
    </div>
    <div
      className="sc-bxivhb sc-EHOje fUTEIU"
    >
      三句
    </div>
  </div>
</div>
`;
